test/ref/test.rs

```diff
@@ -99,8 +118,56 @@ async fn install_canister(agent: &Agent, wasm: &[u8], init_arg: &[u8]) -> Princi
     canister_id
 }
 
-#[tokio::main]
-async fn main() {
+fn unzip_file(file_path: &str, output_path: &str) {
```

Comment:
I tried to do it with bazel first but I didn't manage. The two ways i tried were:
- Use https_archive: Unzips the file but it cannot be set to executable. To make it executable I tried gen_rule. I could however not use gen_rule since the bazel directory is read-only. So the only way would have been to create a new directory in bazel, copy the unzipped binary to that directory and then make it executable through a file group. 
- Use https_file: The file can be set to executable but is not unzipped. Unzipping the file poses the same issue as with https_archive which is that the bazel directory is read-only. 

I opted for this solution as it seemed the most strait forward. Maybe there is an easier solution with bazel that I did not find?

Reply:
Can't we do the same thing we do for replica, sandbox and starter https://sourcegraph.com/github.com/dfinity/ICRC-1/-/blob/bazel/replica_tools.bzl ?